src/lib.rs

```diff
@@ -249,9 +249,15 @@ pub struct RegisterArrayInfo {
 }
 
 #[derive(Clone, Debug)]
+pub struct RegisterClusterInfo {
+    pub registers: Option<Vec<Register>>,
+}
+
+#[derive(Clone, Debug)]
 pub enum Register {
     Single(RegisterInfo),
     Array(RegisterInfo, RegisterArrayInfo),
+    Cluster(RegisterInfo, RegisterClusterInfo),
```

Comment:
Yeah, I was thinking that the type of the `Peripheral.registers` would change to `Vec<Either<Cluster, Register>>`. That would reflect the actual SVD syntax where `<registers>` is an array of `<register>`s and `<cluster>`s.

> I am wondering then if it makes sense to create a new enum

We can use [either](https://crates.io/crates/either)